src/utils/keyExtractor.ts

```diff
@@ -0,0 +1,17 @@
+type Item = any;
+
+export function keyExtractor(item: Item, index: number, extractor?: (item: Item, index: number) => string) {
+  if (extractor != null) {
+    const key = extractor(item, index);
+    if (key != null) {
+      return key;
+    }
+  }
+  if (item.key != null) {
+    return item.key;
+  }
+  if (item.id != null) {
+    return item.id;
+  }
+  return String(index);
+}
```

Comment:
```suggestion
export function defaultKeyExtractor(item: Item, index: number) {
  return index;
}
```

이 함수가 크게 두 가지 역할을 하고 있는데,

1. extractor가 존재하는지 여부를 판단하고 존재하면 extractor를 사용해서 key를 반환
2. 존재하지 않으면 커스텀 extractor를 구현하여 key를 리턴

이 두 가지는 별도의 함수로 각각 존재하는 것이 좋겠음

근데 2번처럼 커스텀 extractor를 구현하면 사용하는 측에서 key로 index를 그냥 쓰고 싶었는데 item에 id나 key가 존재해서 예상대로 동작하지 않는 경우도 생길 수 있기 때문에, 기본 동작으로는 index를 리턴하게 하는 쪽이 좋을 거 같음..! (더 strict하게 간다면 아예 key를 넣지 않아서 개발자에게 key가 없다는 걸 알려주거나, GridList에서 keyExtractor를 필수로 할 수도 있고...)

그래서 결국 `return index;`하는 구문만 남았다고 한다

```
({
className, 
divAttributes,
items,
keyExtractor = defaultKeyExtractor
...
})
...
  <GridListItem key={keyExtractor(item, index)} 
...
```
이런 식으로 하면 될듯



---

- key를 string으로 변환하지 않아도 될듯
react 코드 내에서 key를 항상 string으로 변환하더라구! 그래서 우리 코드에선 깔끔하게 number 그대로 넘겨도 될듯

<img width="419" alt="image" src="https://user-images.githubusercontent.com/5275823/79825981-76a4b580-83d5-11ea-9024-666a28ea2998.png">
https://github.com/facebook/react/blob/0960b7ba580b620202f7d245bb696aca71c63e37/packages/react/src/ReactElement.js#L291